### ISSUE TITLE
Missing `core/post-author-name` in WP Core

### ISSUE THREAD

**Issue by bgturner** [WordPress/gutenberg#44618]
### Description

The Twenty Twenty Three theme uses the `core/post-author-name` block, which isn't included in the`6.1-beta2` version of WP Core.

Specifically, when I use the site editor to edit a single post, I get this error:

```
Your site doesn’t include support for the "core/post-author-name" block. You can leave this block intact or remove it entirely.
```

Installing and activating the Gutenberg plugin removes that error.

### Step-by-step reproduction instructions

1. On a blank plain `6.1-beta2` version of WP, open the site editor
2. Navigate to the "Single" template
3. Scroll to the bottom of the post content and note the missing `core/post-author-name`
4. Install and activate the Gutenberg plugin
5. Return to the "Single" template in the Site editor and note that the error no longer is there.

### Screenshots, screen recording, code snippet

![Image 2022-09-30 at 9 52 17 AM](https://user-images.githubusercontent.com/299432/193319358-d0c56a74-0224-47c4-8ace-3e1adff1d3ca.jpg)

### Environment info

Because this is testing Beta Core, I didn't use Gutenberg initially. Here's the content of my site environment:

```
### wp-core ###

version: 6.1-beta2
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%postname%/
https_status: false
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: local
user_count: 3
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public
wordpress_size: 49.93 MB (52355561 bytes)
uploads_path: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content/uploads
uploads_size: 45.40 MB (47601470 bytes)
themes_path: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content/themes
themes_size: 13.94 MB (14616248 bytes)
plugins_path: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content/plugins
plugins_size: 501.59 KB (513629 bytes)
database_size: 3.19 MB (3342336 bytes)
total_size: 112.94 MB (118429244 bytes)

### wp-active-theme ###

name: Twenty Twenty-Three (twentytwentythree)
version: 1.0
author: the WordPress team
author_website: https://wordpress.org
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, block-templates, widgets-block-editor
theme_path: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content/themes/twentytwentythree
auto_update: Disabled

### wp-themes-inactive (3) ###

Twenty Twenty: version: 2.0, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 1.6, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Two: version: 1.2, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (2) ###

WordPress Beta Tester: version: 3.2.2, author: Peter Westwood, Andy Fragen (latest version: 3.2.3), Auto-updates disabled
WordPress Importer: version: 0.8, author: wordpressdotorg, Auto-updates disabled

### wp-plugins-inactive (1) ###

Akismet Anti-Spam: version: 5.0, author: Automattic (latest version: 5.0.1), Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1808
imagemagick_version: ImageMagick 7.1.0-46 Q16 x86_64 5ef3d4d66:20220816 https://imagemagick.org
imagick_version: 3.7.0
file_uploads: File uploads is turned off
post_max_size: 1000M
upload_max_filesize: 300M
max_effective_size: 300 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 64 GB
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 6144
	imagick::RESOURCETYPE_MAP: 64 GB
	imagick::RESOURCETYPE_MEMORY: 32 GB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: Unable to determine
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, WebP, BMP
ghostscript_version: not available

### wp-server ###

server_architecture: Darwin 21.6.0 x86_64
httpd_software: nginx/1.16.0
php_version: 8.1.9 64bit
php_sapi: fpm-fcgi
max_input_variables: 4000
time_limit: 1200
memory_limit: 256M
max_input_time: 600
upload_max_filesize: 300M
php_post_max_size: 1000M
curl_version: 7.84.0 (SecureTransport) OpenSSL/1.1.1q
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: false

### wp-database ###

extension: mysqli
server_version: 8.0.16
client_version: mysqlnd 8.1.9
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content
WP_PLUGIN_DIR: /Users/ben.turner/Local Sites/wordpress-beta-tester-819/app/public/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: local
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
I can confirm. @carolinan it does not seem like the Post Author Name block was included in 6.1. Any idea why?

**Comment by colorful-tones** [comment]
Confirmed this bug as well, and it will currently impact Twenty Twenty-Three: https://github.com/WordPress/twentytwentythree/blob/trunk/patterns/post-meta.php#L52

**Comment by colorful-tones** [comment]
I think part of the issue here is that Twenty Twenty-Three should likely be using the `core/post-author` block in the `patterns/post-meta.php`. As the `core/post-author-name` is likely an inner child of `core/post-author`.

**Comment by carolinan** [comment]
> I can confirm. @carolinan it does not seem like the Post Author Name block was included in 6.1. Any idea why?

No idea. I am not part of the release squad. I have not seen any conversations about it not being included.

**Comment by carolinan** [comment]
> I think part of the issue here is that Twenty Twenty-Three should likely be using the `core/post-author` block in the `patterns/post-meta.php`. As the `core/post-author-name` is likely an inner child of `core/post-author`.

It is not a child block. The individual blocks were intended to replace post-author but it hasn't been done yet.

**Comment by carolinan** [comment]
Looks like it was made experimental 5 months ago @gziolo @adamziel.

**Comment by ndiego** [comment]
cc @mikachan

**Comment by mikachan** [comment]
Thanks for the ping! @colorful-tones fixed this in TT3 by using the post-author block instead of post-author-name: https://github.com/WordPress/twentytwentythree/pull/236

It would still be good to follow-up with more details around the post-author-name block, but we can probably close this issue based on this being fixed in TT3 directly (if this is the only thing affected by this block).

**Comment by ndiego** [comment]
Thanks @mikachan. @gziolo @adamziel @ockham what are the steps needed to make this block not experimental? Since we now have a fix for TT3, and the addition of a new block is not within the scope of a Beta/RC, the addition of the block to Core will need to get punted to 6.2. That said, I want to make sure we don't miss it again. Anecdotally, I know this block is being used quite a bit in experimental block themes.

**Comment by ockham** [comment]
I think the `_experimental` flag needs to be removed from `block.json`:

https://github.com/WordPress/gutenberg/blob/86730151579447a6c298cccfb7401b31c067fddc/packages/block-library/src/post-author-name/block.json#L4

(I think we used to have an allow-list of blocks in Core, but that changed per @gziolo's https://github.com/WordPress/wordpress-develop/pull/2940 which now evaluates that `block.json` field instead.)

**Comment by ndiego** [comment]
Now that TT3 has been updated to not require the Post Author Name block, there is no urgency on this issue for 6.1 but will target 6.2 for inclusion.

**Comment by desrosj** [comment]
Could someone clarify whether the intention was to include the `core/post-author-name` block as stable and not experimental in 6.1? Or, are there are any other side effects elsewhere in the editor for 6.1 as a result of this block not being marked stable and available at core/post-author-name?

I'm trying to determine if #45008 is something that should be merged and included in 6.1.1.

**Comment by ockham** [comment]
I think the original intention was to include it as stable, but it didn't make it in time before the cut-off. TT3 was fixed so it'd work without the block.
